File: types/command.ts
import { LayerStatus } from './index';

export type OperationalSeverity = 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW' | 'INFO';

export type OperationalEventCategory =
  | 'MARITIME SAFETY'
  | 'BOUNDARY'
  | 'ENVIRONMENTAL'
  | 'PFZ'
  | 'SATELLITE'
  | 'SYSTEM';

export type OperationalStatus = 'NEW' | 'ACKNOWLEDGED' | 'INVESTIGATING' | 'RESOLVED';

export interface OperationalEvent {
  id: string;
  category: OperationalEventCategory;
  severity: OperationalSeverity;
  status: 'DEMO';
  workflowStatus: OperationalStatus;
  title: string;
  description: string;
  latitude: number;
  longitude: number;
  locationName: string;
  timestamp: string;
  source: string;
  product?: string;
  dataset?: string;
  variable?: string;
  dataStatus: 'DEMO' | 'REAL DATA' | 'UNAVAILABLE';
  metadata?: {
    vesselName?: string;
    vesselCallsign?: string;
    distanceKm?: number;
    pfzZoneId?: string;
    platformId?: string;
    anomalyMetric?: string;
  };
}

export interface DemoVessel {
  id: string;
  name: string;
  callsign: string;
  flag: string;
  type: string;
  latitude: number;
  longitude: number;
  heading: number; // degrees
  speedKnots: number;
  status: 'DEMO';
  track: [number, number][]; // [lng, lat]
}

export interface SystemServiceStatus {
  serviceId: string;
  name: string;
  status: LayerStatus;
  latencyMs?: number;
  description: string;
}

File: backend/app/schemas/command.py
"""
ORCA Backend — Command Center Schemas
Aligned with command.ts TypeScript interfaces and mockCommand.ts shapes.
"""
from typing import Any, Dict, List, Optional

from pydantic import BaseModel


class OperationalEventSchema(BaseModel):
    id: str
    category: str  # WEATHER | VESSEL | BOUNDARY | ENVIRONMENTAL | SYSTEM
    severity: str  # CRITICAL | HIGH | MEDIUM | LOW | INFO
    status: str = "DEMO"
    workflow_status: str  # NEW | ACKNOWLEDGED | INVESTIGATING | RESOLVED
    title: str
    description: str
    latitude: float
    longitude: float
    location_name: str
    timestamp: str
    source: str
    data_status: str
    metadata: Optional[Dict[str, Any]] = None


class VesselSchema(BaseModel):
    id: str
    name: str
    type: str
    latitude: float
    longitude: float
    heading: float  # degrees
    speed: float  # knots
    status: str
    last_updated: str
    flag: Optional[str] = None
    mmsi: Optional[str] = None
    track: Optional[List[List[float]]] = None  # [[lat, lng], ...]


class SystemServiceStatus(BaseModel):
    name: str
    status: str  # ONLINE | DEGRADED | OFFLINE
    latency_ms: Optional[float] = None
    last_update: Optional[str] = None
    data_status: str = "DEMO"


class EventStatusUpdateRequest(BaseModel):
    status: str  # ACKNOWLEDGED | INVESTIGATING | RESOLVED
